lib/ConvoyFollower/src/DrivingState.h

```diff
@@ -156,17 +150,89 @@ class DrivingState : public IState
         return m_isActive;
     }
 
+    /**
+     * Get the current average Inter Vehicle Space (IVS) in mm.
+     *
+     * @return Current average IVS in mm.
+     */
+    int32_t getAvgIVS() const
+    {
+        return m_avgIvs.getResult();
+    }
+
 protected:
 private:
     /** Maximum invalid waypoints allowed before going into error state. */
     static const uint8_t MAX_INVALID_WAYPOINTS = 3U;
 
+    /** Aperture angle of the forward cone in mrad. */
+    static const int32_t FORWARD_CONE_APERTURE = 1300; /* Aprox. 75 degrees */
+
+    /** Period in ms for PID processing. */
+    static const uint32_t IVS_PID_PROCESS_PERIOD = 50U;
+
+    /** Default Inter Vehicle Space in mm. */
+    static const int32_t DEFAULT_IVS = 200;
+
+    /**
+     * Number of measurements to be taken into account in the average IVS.
+     * Calculated as the number of measurements in a second.
+     */
+    static const uint8_t IVS_MOVAVG_SIZE = 1000U / IVS_PID_PROCESS_PERIOD;
+
+    /**
+     * Error margin in mm for target waypoint.
+     * Used to determine if target waypoint has been reached.
+     */
+    static const int32_t TARGET_WAYPOINT_ERROR_MARGIN = 30;
+
+    /** PID factors for the Inter Vehicle Space Controller. */
+    struct IVS_PID_FACTORS
+    {
+        /** The PID proportional factor numerator for the Inter Vehicle Space Controller. */
+        static const int32_t PID_P_NUMERATOR = 3;
+
+        /** The PID proportional factor denominator for the Inter Vehicle Space Controller.*/
+        static const int32_t PID_P_DENOMINATOR = 4;
+
+        /** The PID integral factor numerator for the Inter Vehicle Space Controller. */
+        static const int32_t PID_I_NUMERATOR = 1;
+
+        /** The PID integral factor denominator for the Inter Vehicle Space Controller. */
+        static const int32_t PID_I_DENOMINATOR = 10;
+
+        /** The PID derivative factor numerator for the Inter Vehicle Space Controller. */
+        static const int32_t PID_D_NUMERATOR = 1;
+
+        /** The PID derivative factor denominator for the Inter Vehicle Space Controller. */
+        static const int32_t PID_D_DENOMINATOR = 10;
+    };
+
+    /** PID factors for the Heading. */
+    struct HEADING_FINDER_PID_FACTORS
+    {
+        /** The PID proportional factor numerator for the Heading Finder. */
+        static const int32_t PID_P_NUMERATOR = 2;
+
+        /** The PID proportional factor denominator for the Heading Finder.*/
+        static const int32_t PID_P_DENOMINATOR = 1;
+
+        /** The PID integral factor numerator for the Heading Finder. */
+        static const int32_t PID_I_NUMERATOR = 0;
+
+        /** The PID integral factor denominator for the Heading Finder. */
+        static const int32_t PID_I_DENOMINATOR = 1;
+
+        /** The PID derivative factor numerator for the Heading Finder. */
+        static const int32_t PID_D_NUMERATOR = 100;

```

Comment:
wow ... sounds high ...

Reply:
It is...but it works :D
Optimization of the PID controllers will be done in the next PR